h2/settings.py

```diff
@@ -253,7 +253,18 @@ def __iter__(self):
     def __len__(self):
         return len(self._settings)
 
-
+    def __eq__(self, other):
+        if isinstance(other, Settings):
+            return self == other
```

Comment:
So, this doesn't work because it leads to a recursive definition. Specifically, it says `self` is equal to `other` if `self` equals `other`. That's tautologically accurate, but doesn't help Python to actually work out what equality looks like.

This should be changed to specifically compare the `_settings` properties of the two objects.